ServiceWorkerIT::offlineStub_backOnline_stubRemoved_serverViewShown is flaky in Bender

### Description of the bug

ServiceWorkerIT::offlineStub_backOnline_stubRemoved_serverViewShown is flaky in Bender.

<img width="1594" alt="Screenshot 2022-08-17 at 9 36 45" src="https://user-images.githubusercontent.com/61410877/185051217-cce89be1-f55a-49f9-a54c-3d70f055403f.png">

```
java.lang.RuntimeException: offlineStub_backOnline_stubRemoved_serverViewShown[ANY_Chrome_](com.vaadin.flow.navigate.ServiceWorkerIT): run failed 2 times
	at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:78)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:410)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.openqa.selenium.TimeoutException: Expected condition failed: waiting for com.vaadin.flow.testutil.TestBenchHelpers$$Lambda$778/0x0000000100535440@212b9efc (tried for 10 second(s) with 500 milliseconds interval)
Build info: version: '4.4.0', revision: 'e5c75ed026a'
System info: host: 'ip-172-16-1-193.eu-west-1.compute.internal', ip: '172.16.1.193', os.name: 'Linux', os.arch: 'amd64', os.version: '4.14.268-205.500.amzn2.x86_64', java.version: '11.0.14.1'
Driver info: com.vaadin.testbench.TestBenchDriverProxy_$$_jvst5f8_0
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 104.0.5112.79, chrome: {chromedriverVersion: 104.0.5112.79 (3cf3e8c8a07d..., userDataDir: /tmp/.com.google.Chrome.vJWsVF}, goog:chromeOptions: {debuggerAddress: localhost:40053}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: LINUX, proxy: Proxy(), se:cdp: ws://validation-hub.rnd.aws..., se:cdpVersion: 104.0.5112.79, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: aec97f64ef53dbfe131148f8abd7d60f109f9e3445d3115a70528b387d18340b
	at org.openqa.selenium.support.ui.WebDriverWait.timeoutException(WebDriverWait.java:87)
	at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:231)
	at com.vaadin.testbench.TestBenchTestCase.waitUntil(TestBenchTestCase.java:223)
	at com.vaadin.testbench.TestBenchTestCase.waitUntil(TestBenchTestCase.java:247)
	at com.vaadin.flow.testutil.TestBenchHelpers.waitForElementNotPresent(TestBenchHelpers.java:140)
	at com.vaadin.flow.navigate.ServiceWorkerIT.offlineStub_backOnline_stubRemoved_serverViewShown(ServiceWorkerIT.java:276)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at com.vaadin.testbench.parallel.ParallelRunner$TBMethod.invokeExplosively(ParallelRunner.java:492)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at com.vaadin.testbench.parallel.ParallelRunner$1.evaluate(ParallelRunner.java:466)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:68)
	... 12 more
------- Stderr: -------
Aug 16, 2022 11:24:52 PM org.openqa.selenium.remote.RemoteWebDriver checkNonW3CCapabilities
WARNING: Support for Legacy Capabilities is deprecated; You are sending the following invalid capabilities: [networkConnectionEnabled]; Please update to W3C Syntax: https://www.selenium.dev/blog/2022/legacy-protocol-support/
Aug 16, 2022 11:24:52 PM org.openqa.selenium.remote.RemoteWebDriver checkNonW3CCapabilities
WARNING: Support for Legacy Capabilities is deprecated; You are sending the following invalid capabilities: [networkConnectionEnabled]; Please update to W3C Syntax: https://www.selenium.dev/blog/2022/legacy-protocol-support/
Aug 16, 2022 11:24:54 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected upstream dialect: W3C
Aug 16, 2022 11:24:54 PM org.openqa.selenium.remote.ProtocolHandshake createSession
WARNING: Support for Legacy Capabilities is deprecated; You are sending the following invalid capabilities: [networkConnectionEnabled]; Please update to W3C Syntax: https://www.selenium.dev/blog/2022/legacy-protocol-support/
Aug 16, 2022 11:25:10 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:10 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:10 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:21 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:21 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:21 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:21 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:21 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:21 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
[pool-1-thread-4] INFO com.vaadin.testbench.ScreenshotOnFailureRule - Error screenshot written to: /opt/agent/work/1e3f75bb7ff3ed55/flow-tests/test-ccdm-flow-navigation/error-screenshots/offlineStub_backOnline_stubRemoved_serverViewShown[ANY_Chrome_](com.vaadin.flow.navigate.ServiceWorkerIT).png
Aug 16, 2022 11:25:22 PM org.openqa.selenium.remote.RemoteWebDriver checkNonW3CCapabilities
WARNING: Support for Legacy Capabilities is deprecated; You are sending the following invalid capabilities: [networkConnectionEnabled]; Please update to W3C Syntax: https://www.selenium.dev/blog/2022/legacy-protocol-support/
Aug 16, 2022 11:25:23 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected upstream dialect: W3C
Aug 16, 2022 11:25:23 PM org.openqa.selenium.remote.ProtocolHandshake createSession
WARNING: Support for Legacy Capabilities is deprecated; You are sending the following invalid capabilities: [networkConnectionEnabled]; Please update to W3C Syntax: https://www.selenium.dev/blog/2022/legacy-protocol-support/
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:24 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:34 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:34 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:34 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:34 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:34 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
Aug 16, 2022 11:25:34 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 104
[pool-1-thread-4] INFO com.vaadin.testbench.ScreenshotOnFailureRule - Error screenshot written to: /opt/agent/work/1e3f75bb7ff3ed55/flow-tests/test-ccdm-flow-navigation/error-screenshots/offlineStub_backOnline_stubRemoved_serverViewShown[ANY_Chrome_](com.vaadin.flow.navigate.ServiceWorkerIT).png
```

Logs:
[Uploading Flow_23.2_Flow_23.2_Snapshot_master_926.log…]()

### Expected behavior

Test passes always

### Minimal reproducible example

See Flow 23.2 snapshot validation build

### Versions

- Vaadin / Flow version: 23.2.0.beta1

--- a/flow-client/src/main/frontend/Flow.ts
+++ b/flow-client/src/main/frontend/Flow.ts
@@ -386,7 +386,9 @@ export class Flow {
         http.onerror = () => {
           $wnd.Vaadin.connectionState.state = ConnectionState.CONNECTION_LOST;
         };
-        http.send();
+        // Postpone request to reduce potential net::ERR_INTERNET_DISCONNECTED
+        // errors that sometimes occurs even if browser says it is online
+        setTimeout(() => http.send(), 50);
       }
     });
     $wnd.addEventListener('offline', () => {

--- a/flow-tests/test-ccdm-flow-navigation/src/test/java/com/vaadin/flow/navigate/ServiceWorkerIT.java
+++ b/flow-tests/test-ccdm-flow-navigation/src/test/java/com/vaadin/flow/navigate/ServiceWorkerIT.java
@@ -262,9 +262,7 @@ public class ServiceWorkerIT extends ChromeDeviceTest {
 
     @Test
     public void offlineStub_backOnline_stubRemoved_serverViewShown() {
-        getDriver().get(getRootURL() + "/");
-        waitForDevServer();
-        waitForServiceWorkerReady();
+        openPageAndPreCacheWhenDevelopmentMode("/");
         getDevTools().setOfflineEnabled(true);
 
         try {

